core/job/service/plugin_service.go

```diff
@@ -121,9 +121,12 @@ func (p JobPluginService) GenerateDependencies(ctx context.Context, taskName job
 		return nil, err
 	}
 
-	// flatten
+	// flatten and ignore the destinatioon
 	resourceURNs := []job.ResourceURN{}
 	for _, r := range job.ResourceURNWithUpstreamsList(resources).Flatten() {
+		if r.URN == destinationURN {
+			continue
```

Comment:
how about we log such a scenarion, so that we can see , what all diffrent clauses are causing this

Reply:
forgot to add it, yes we could log it as a warning